cpu/numactl.py

```diff
@@ -67,6 +72,57 @@ def setUp(self):
 
         build.make(self.sourcedir)
 
+        self.iface = self.params.get("interface", default="")
+        self.ping_count = self.params.get("ping_count", default=100)
+        self.peer = self.params.get("peer_ip", default="")
+        interfaces = netifaces.interfaces()
+        if not self.iface:
+            self.cancel("Please specify interface to be used")
+        if self.iface not in interfaces:
+            self.cancel("%s interface is not available" % self.iface)
+        if not self.peer:
+            self.cancel("peer ip should specify in input file")
+        self.ipaddr = self.params.get("host_ip", default="")
+        self.netmask = self.params.get("netmask", default="")
+        self.localhost = LocalHost()
+        self.networkinterface = NetworkInterface(self.iface, self.localhost)
+        try:
+            self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+            self.networkinterface.save(self.ipaddr, self.netmask)
+        except Exception:
+            self.networkinterface.save(self.ipaddr, self.netmask)
+        self.networkinterface.bring_up()
+        if not wait.wait_for(self.networkinterface.is_link_up, timeout=60):
+            self.cancel("Link up of interface taking longer than 60 seconds")
+        if self.networkinterface.ping_check(self.peer, count=5) is not None:
+            self.cancel("No connection to peer")
+        self.device = self.params.get('pci_device', default="")
+        if not self.device:
+            self.cancel("PCI_address not given")
+        if not os.path.isdir('/sys/bus/pci/devices/%s' % self.device):
+            self.cancel("%s not present in device path" % self.device)
+        self.cpu_path = "/sys/devices/system/node/has_cpu"
+        if not os.path.exists(self.cpu_path):
+            self.cancel("No NUMA nodes have CPU")
+        self.numa_dict = cpu.numa_nodes_with_assigned_cpus()
+
+    def check_numa_nodes(self):
+        if len(cpu.numa_node_has_cpus()) < 2:
+            self.cancel("Required atleast two NUMA nodes with CPU"
+                        " assigned for this test case!")
+        else:
+            return True
+
+    def numa_ping(self, cmd):
+        '''
+        Ping Test to remote Peer with -f and count options.
+        '''
+        output = process.run(cmd, shell=True,
+                             ignore_status=True
+                             ).stdout.decode("utf-8").split(",")
+        if " 0% packet loss" not in output:
+            self.cancel("failed due to packet loss")
```

Comment:
why not push this as part of utility itself

Reply:
pandey,  we did explore pushing to exiting ping utils.. but we found not feasible.. also this is not being used outside this test so this will be here only